nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/dto/EntityFactory.java

```diff
@@ -77,6 +79,27 @@
 
 public final class EntityFactory {
 
+    public ProcessorDiagnosticsEntity createProcessorDiagnosticsEntity(final ProcessorDiagnosticsDTO dto, final RevisionDTO revision, final PermissionsDTO processorPermissions,
+        final ProcessorStatusDTO status, final List<BulletinEntity> bulletins) {
+        final ProcessorDiagnosticsEntity entity = new ProcessorDiagnosticsEntity();
+        entity.setRevision(revision);
+        if (dto != null) {
+            entity.setPermissions(processorPermissions);
+            entity.setId(dto.getProcessor().getId());
+            if (processorPermissions != null && processorPermissions.getCanRead()) {
+                entity.setComponent(dto);
+                entity.setBulletins(bulletins);
+            }
+        }
+
+        entity.setBulletins(bulletins);
+        return entity;
+    }
+
+    private void pairDownDiagnostics(final ProcessorDiagnosticsDTO dto, final PermissionsDTO controllerPermissions) {
```

Comment:
This is method OBE since the permission-based filtering happens in the service facade?

Reply:
Whoops, yes, good call.